crates/scroll/chainspec/src/genesis.rs

```diff
@@ -0,0 +1,214 @@
+//! Scroll types for genesis data.
+
+use alloy_primitives::Address;
+use alloy_serde::OtherFields;
+use serde::de::Error;
+
+/// Container type for all Scroll specific fields in a genesis file.
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ScrollChainInfo {
+    /// Genesis information
+    pub genesis_info: Option<ScrollGenesisInfo>,
+    /// scroll chain special information
+    pub scroll_special_info: Option<ScrollSpecialChainInfo>,
+}
+
+impl ScrollChainInfo {
+    /// Extracts the Scroll specific fields from a genesis file. These fields are expected to be
+    /// contained in the `genesis.config` under `extra_fields` property.
+    pub fn extract_from(others: &OtherFields) -> Option<Self> {
+        Self::try_from(others).ok()
+    }
+}
+
+impl TryFrom<&OtherFields> for ScrollChainInfo {
+    type Error = serde_json::Error;
+
+    fn try_from(others: &OtherFields) -> Result<Self, Self::Error> {
+        let genesis_info = ScrollGenesisInfo::try_from(others).ok();
+        let scroll_special_info = ScrollSpecialChainInfo::try_from(others).ok();
+
+        Ok(Self { genesis_info, scroll_special_info })
+    }
+}
+
+/// The Scroll-specific genesis block specification.
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ScrollGenesisInfo {
+    /// archimedes block number
+    pub archimedes_block: Option<u64>,
+    /// bernoulli block number
+    pub bernoulli_block: Option<u64>,
+    /// curie block number
+    pub curie_block: Option<u64>,
+    /// darwin hardfork timestamp
+    pub darwin_time: Option<u64>,
+    /// darwinV2 hardfork timestamp
+    pub darwin_v2_time: Option<u64>,
+}
+
+impl ScrollGenesisInfo {
+    /// Extract the Scroll-specific genesis info from a genesis file.
+    pub fn extract_from(others: &OtherFields) -> Option<Self> {
+        Self::try_from(others).ok()
+    }
+}
+
+impl TryFrom<&OtherFields> for ScrollGenesisInfo {
+    type Error = serde_json::Error;
+
+    fn try_from(others: &OtherFields) -> Result<Self, Self::Error> {
+        others.deserialize_as()
+    }
+}
+
+/// The Scroll l1 special config
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct L1Config {
+    /// l1 chain id
+    pub l1_chain_id: u64,
+    /// the l1 message queue address
+    pub l1_message_queue_address: Address,
+    // the l1 scroll proxy address
+    pub scroll_chain_address: Address,
+    // the l1 message numbers of per block
+    pub num_l1_messages_per_block: u64,
+}
+
+/// The Scroll special chain specification.
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ScrollSpecialChainInfo {
+    /// the L2 tx fee vault address
+    pub fee_vault_address: Option<Address>,
```

Comment:
The fee_vault_address is optional, can refer here: [1](https://github.com/scroll-tech/go-ethereum/blob/develop/params/config.go#L606) [2](https://github.com/scroll-tech/go-ethereum/blob/develop/core/evm.go#L48). 

But the L1Config is necessary for Rollup

Reply:
Lets add some comments here. `If the fee_vault_address is provided it will act as the beneficiary and receive transaction fees and block rewards. If no fee_vault_address is provided then the block author will be used as the beneficiary`. 

If `L1Config` is required then it should not be an option?

Reply:
yes, i'll change it

Reply:
fixed [0cee1ba](https://github.com/scroll-tech/reth/pull/56/commits/0cee1ba025668a3f68e526c138fc08bad4f2aeff)